### PR TITLE
Fixes to running mocha in parallel.

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by MathieuAA** [comment]
Feel free to merge it and close mine. Glad you fixed those!!!

**Comment by mshima** [comment]
@MathieuAA can you test it, I am currently with a dual-core processor.

**Comment by MathieuAA** [comment]
Let me check...

**Comment by MathieuAA** [comment]
@mshima execution time: 5 minutes.
I got two errors, which is weird:
```
  1) jhipster:app --incremental-changelog
       when removing a field with constraints
         "before all" hook for "should create application":
     Error: Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/mathieu/WebstormProjects/generator-jhipster/test/app/incremental-changelog.spec.js)
      at listOnTimeout (internal/timers.js:549:17)
      at processTimers (internal/timers.js:492:7)

  2) jhipster:app --incremental-changelog
       when removing a field with constraints
         "after all" hook for "should create the entity constraint update changelog":
     TypeError: Cannot read property 'cleanup' of undefined
      at Context.<anonymous> (test/app/incremental-changelog.spec.js:533:31)
      at processImmediate (internal/timers.js:456:21)
```

**Comment by mshima** [comment]
First error seems a normal timeout. Probably due to the resources been used in parallel.
Increasing timeout for this test should fix it.
Second is due to to first, I need to fix yeoman-test to make after step independent of before.

File: test/app/incremental-changelog.spec.js [hunk 1]
```diff
@@ -36,7 +36,8 @@ relationship ManyToOne {
 }
 `;
 
-describe('jhipster:app --incremental-changelog', () => {
+describe('jhipster:app --incremental-changelog', function() {
```
Comment (mshima) [inline test/app/incremental-changelog.spec.js:39]:
```suggestion
describe('jhipster:app --incremental-changelog', function () {
```

**Comment by pascalgrimaud** [comment]
just tested and it works well, good job @mshima

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**